polkadot/xcm/xcm-builder/src/routing.rs

```diff
@@ -149,12 +153,21 @@ impl EnsureDelivery for Tuple {
 /// Inspects messages in queues.
 /// Meant to be used in runtime APIs, not in runtimes.
 pub trait InspectMessageQueues {
+	/// Clear the queues at the beginning of Runtime API call, so that subsequent
+	/// `Self::get_messages()` will return only messages generated by said Runtime API.
+	fn clear_messages();
```

Comment:
the backport should have default impl here to not be breaking - runtimes that consume this patch release can overwrite it if they want